onnxruntime/core/platform/windows/telemetry.cc

```diff
@@ -253,27 +253,52 @@ void WindowsTelemetry::LogSessionCreation(uint32_t session_id, int64_t ir_versio
     execution_provider_string += i;
   }
 
-  TraceLoggingWrite(telemetry_provider_handle,
-                    "SessionCreation",
-                    TraceLoggingBool(true, "UTCReplace_AppSessionGuid"),
-                    TelemetryPrivacyDataTag(PDT_ProductAndServiceUsage),
-                    TraceLoggingKeyword(MICROSOFT_KEYWORD_MEASURES),
-                    TraceLoggingKeyword(static_cast<uint64_t>(onnxruntime::logging::ORTTraceLoggingKeyword::Session)),
-                    TraceLoggingLevel(WINEVENT_LEVEL_INFO),
-                    // Telemetry info
-                    TraceLoggingUInt8(0, "schemaVersion"),
-                    TraceLoggingUInt32(session_id, "sessionId"),
-                    TraceLoggingInt64(ir_version, "irVersion"),
-                    TraceLoggingUInt32(projection_, "OrtProgrammingProjection"),
-                    TraceLoggingString(model_producer_name.c_str(), "modelProducerName"),
-                    TraceLoggingString(model_producer_version.c_str(), "modelProducerVersion"),
-                    TraceLoggingString(model_domain.c_str(), "modelDomain"),
-                    TraceLoggingBool(use_fp16, "usefp16"),
-                    TraceLoggingString(domain_to_verison_string.c_str(), "domainToVersionMap"),
-                    TraceLoggingString(model_graph_name.c_str(), "modelGraphName"),
-                    TraceLoggingString(model_metadata_string.c_str(), "modelMetaData"),
-                    TraceLoggingString(loaded_from.c_str(), "loadedFrom"),
-                    TraceLoggingString(execution_provider_string.c_str(), "executionProviderIds"));
+  // Difference is MeasureEvent & isCaptureState, but keep in sync otherwise
+  if (!captureState) {
+    TraceLoggingWrite(telemetry_provider_handle,
+                      "SessionCreation",
```

Comment:
Particular reason different event name was chosen instead of a bool indicating capture state? Not a major issue but there tracelogging does have a not trivial impact on stack space.

Reply:
In this case I really did not like using the event name of SessionCreation as it might be misleading for most folks in the common case. They would have to know about and look for the isCaptureState flag. We use the flag elsewhere for logging options where it doesn’t seem like a big deal whether it’s at session creation or captureState. 

In addition the original SessionCreation is a measureEvent. Didn’t want to add an unnecessary flag there for normal actual SessionCreation

Reply:
Would just calling "SessionInfo" make it less confusing? And then bool or some other type of field clarifying if it was a rundown or not. Given the expected low frequency of "manual" rundown calls, even those could be MEASURES keyword...or if really need be the keyword can be toggled at runtime for the same event.

In general, when there is a tradeoff, I'd like to index more on production code efficiency (stack/code size here) than some guidance aspect on viewing the events.

Reply:
I would totally agree with you (mostly) if this was new code. I like your rename suggestion. But this is an existing measure event that likely ORT folks rely on for certain scenarios. I will let them comment if we can break things with a rename. I don't like the rundown being a measure event, since it can lead to overcounting if not careful, but that's also for ORT folks to discuss.  The if/else does lead to slightly larger code size and is not elegant/concise so it that is going against it. We need to balance both needs here.

Reply:
Even when a single event is used (which I recommend here), rundown not being a MEASURES event is solvable by flipping keyword at run-time. I am unclear about tradeoffs on the event analysis workflow with either using a new event name or reusing existing one for the rundown case (given we can have a separate bool for it). 

I don't love adding (more than necessary) overhead in production code in favor of some inconveniences/adjustments to dev analysis process or even existing tools on process events etc. 

Maybe using the existing eventname for rundown case but not logging that case with MEASURES keyword is a good happy medium... We've used that same pattern in other cases in Windows as well.

Reply:
I had some compile issues for MACRO conditionals previously but will try again. What I wanted originally and tried was what you just proposed. Leave the SessionCreation event alone with measures keyword and don't add isRunDown bool flag. But conditionally remove measures keyword and add isRunDown when it's needed for local trace.

Reply:
ORT folks, what are your thoughts on changing the event name? E.g. How much this event is used for a breaking Telemetry change? If we don't change the name we probably need to leave the if/else because ETW macros require compile-time constants